### PR TITLE
pr for microverse tse

### PR DESCRIPTION
**Pull request by RoodzFernando**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  # resources :users, only: [:new, :create]
+  resources :users
```
Comment (wathigo) [inline config/routes.rb]: **[REQUIRED]**
*We need `resources :users, only: [:new, :create, :edit, :update]` in config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to '/users', notice: 'user was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit; end
```
Comment (wathigo) [inline app/controllers/users_controller.rb:25]: **[REQUIRED]**
* Finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  def update; end
```
Comment (wathigo) [inline app/controllers/users_controller.rb]: **[REQUIRED]**
* Finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).
* `redirect_to edit_user_path` if user was saved.
* `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+      <ul>
+        <% @user.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
```
Comment (wathigo) [inline app/views/users/new.html.erb]: **[REQUIRED]**
We need a **commented out** `form` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (wathigo) [inline app/controllers/users_controller.rb:25]: **[REQUIRED]**
* Finds user based on id from params, i.e. @user = User.find(params[:id]) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (wathigo) [inline app/controllers/users_controller.rb]: **[REQUIRED]**
* Finds user based on id from params, i.e. `@user = User.find(params[:id])` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).
* `redirect_to edit_user_path` if user was saved.
* `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <%= form.label :username %>
+    <%= form.text_field :username %>
+  </div>
+
+  <div class="field">
```
Comment (wathigo) [inline app/views/users/new.html.erb]: **[REQUIRED]**
* Contains a commented out `form` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).
* Contains a commented out `form_tag` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)).
* Contains `form_for` that uses `@user` from the controller.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_for.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/views/users/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+  <div class="field">
+    <%= form.label :password %>
+    <%= form.password_field :password %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+<% end %>
```
Comment (wathigo) [inline app/views/users/edit.html.erb:77]: **[REQUIRED]**
* Use `form_for` in `users/edit.html/erb` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_edit.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required 
Hi @RoodzFernando  and @firmoholanda :wave: 
We need to meet some requirements before we move on.
See the comments :point_down: 

**After implementing the changes, submit another request for review.**

**Happy Coding  :computer: 
Simon Wathigo**

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  <% end %>
+
+  <div class="field">
+    <%= f.label :username %><br>
+    <%= f.text_field :username %>
+  </div>
+
+  <div class="field">
+    <%= f.label :email %><br>
+    <%= f.text_area :email %>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: `textarea` is not good for email

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :password %><br>
+    <%= f.text_area :password %>
```
Comment (maelfosso) [inline app/views/users/new.html.erb]: `textarea` is not good for password

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,49 @@
+<!--
+<form action="/users" method="post" accept-charset="UTF-8">
+Username: <input type="text" name="username"><br> <br>
+
+Email: <input type="email" name="email"><br><br>
```
Comment (maelfosso) [inline app/views/users/new.html.erb:5]: Please, I advise you to use `user[username]` and `user[email]`

**Review by maelfosso — CHANGES_REQUESTED** [review]
# changes required :red_circle: 

Hello @RoodzFernando @firmoholanda 

Thank you for the improvements.
However, there are still changes that should be done before you move forward.

1. As @wathigo  said in last review, there is still missing one commented form, the one with `form_tag`.

2.  Read the comments below.

Kindly make the changes and submit a new code review requests.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Comment by RoodzFernando** [comment]
Changes implemented!

File: app/views/users/edit.html.erb [hunk 10]
```diff
@@ -1,3 +1,16 @@
...
+
+Email: <input type="email" name="email"><br><br>
+
+Password: <input type="password" name="password"><br>
+<input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
+  <input type="submit" value="Submit">
+</form>
+-->
+
+
```
Comment (uche-inyama) [inline app/views/users/edit.html.erb]: Hi team!

You have not tried out form_tag in this project. From the instructions, you are to implement a form, with form_tag, then comment it out. 

Kindly look at this [material](https://api.rubyonrails.org/v5.1.7/classes/ActionView/Helpers/FormTagHelper.html) as a reference;  also, read this [material](https://guides.rubyonrails.org/v3.0/form_helpers.html). Scroll through the web pages carefully to see the topic on form_tag. :slightly_smiling_face:

**Review by uche-inyama — CHANGES_REQUESTED** [review]
Hi team! :wave: 

Kindly look at my comment below.

If you need any clarifications, please do reachout on slack @ Inyama, Uchechukwu Henry.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Review by maelfosso — APPROVED** [review]
# Congratulations :+1: :clap: 

You can merge it :smile: 
Good luck with the next step

Thanks
**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request merged**